packages/core/admin/ee/server/seat-enforcement.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+const _ = require('lodash');
+
+const enableUsersToLicenseLimit = async (numberOfUsersToEnable) => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data || !data.value || data.value.length === 0) return;
+
+  const disabledUsers = JSON.parse(data.value);
+
+  const usersToEnable = _.take(disabledUsers, numberOfUsersToEnable);
+
+  usersToEnable.forEach(async (user) => {
+    const data = await strapi.db.query('admin::user').findOne({
+      where: { id: user.id },
+    });
+
+    if (!data) return;
+
+    await strapi.db.query('admin::user').update({
+      where: { id: user.id },
+      data: { isActive: true },
+    });
+  });
+
+  const remainingDisabledUsers = _.drop(disabledUsers, numberOfUsersToEnable);
+
+  await strapi.db.query('strapi::ee-store').update({
+    where: { id: data.id },
+    data: { value: JSON.stringify(remainingDisabledUsers) },
+  });
+};
+
+const calculateAdminSeatDifference = async (seatsAllowedByLicense) => {
+  const currentAdminSeats = await strapi.db.query('admin::user').count({
```

Comment:
We need the service for this one, this is the third time we're fetching the activeAdminUserCount :eyes: